pkg/state/filter.go

```diff
@@ -89,35 +74,51 @@ func filterOutDynamicAttributes(iface map[string]interface{}) {
 	delete(options, "hello-timer")
 }
 
-func filterOutInterfaces(state map[string]interface{}, interfacesFilterGlob glob.Glob) {
-	interfaces := state["interfaces"]
-	filteredInterfaces := []interface{}{}
-
-	for _, iface := range interfaces.([]interface{}) {
-		name := iface.(map[string]interface{})["name"]
-		if !interfacesFilterGlob.Match(name.(string)) {
-			filterOutDynamicAttributes(iface.(map[string]interface{}))
+func filterOutInterfaces(ifacesState []interfaceState, interfacesFilterGlob glob.Glob) []interfaceState {
+	filteredInterfaces := []interfaceState{}
+	for _, iface := range ifacesState {
+		if !interfacesFilterGlob.Match(iface.Name) {
+			filterOutDynamicAttributes(iface.Data)
 			filteredInterfaces = append(filteredInterfaces, iface)
 		}
 	}
-	state["interfaces"] = filteredInterfaces
+	return filteredInterfaces
 }
 
 func filterOut(currentState shared.State, interfacesFilterGlob glob.Glob) (shared.State, error) {
-	var state map[string]interface{}
-	err := yaml.Unmarshal(currentState.Raw, &state)
-	if err != nil {
+	var state rootState
+	if err := yaml.Unmarshal(currentState.Raw, &state); err != nil {
 		return currentState, err
 	}
 
-	filterOutInterfaces(state, interfacesFilterGlob)
-	filterOutRoutes("running", state, interfacesFilterGlob)
-	filterOutRoutes("config", state, interfacesFilterGlob)
+	if err := normalizeInterfacesNames(currentState.Raw, &state); err != nil {
+		return currentState, err
+	}
```

Comment:
Looks like the `gopkg.in/yaml.v2` has similar functionalit to `MarshalJSON` and `UnmarshalJSON` 

https://pkg.go.dev/gopkg.in/yaml.v2#Marshaler
https://pkg.go.dev/gopkg.in/yaml.v2#Unmarshaler

So using those we can replace totally `sigs.k8s.io/yaml` with `gopkg.in/yaml.v2` so we have the normalization for free and we can also copy the rest of the attributes on those functions.

Reply:
It has different behaviors and produces in some cases `null`. But the main problem is that we need to replace it all over the project.
I tried to do this under filtering but got some nasty issues.

I suggest exploring this on a separate PR.